### PR TITLE
[DEV-1896] Connect tutorial list to page

### PR DESCRIPTION
**Pull request by MarcoPonchia**
Depends on: #1077 

#### List of Changes

- Fetch TutorialListPages from Strapi
- Show TutorialListPages from Strapi if found or static as fallback

#### Motivation and Context

Editor can manage TutorialListPage content from Strapi

#### How Has This Been Tested?

manually (to proper test this PR a TutorialListPage must be created and published in Strapi)

#### Screenshots (if appropriate):
screenshot-localhost_3000-2024 09 03-17_36_52
Screenshot from 2024-09-03 17-37-10

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .changeset/polite-bats-own.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"nextjs-website": minor
+---
+
+Connect product tutorial list page to strapi content
```
Comment (marcobottaro) [inline .changeset/polite-bats-own.md:5]:
```suggestion
Connect product tutorial list page to Strapi content
```

File: apps/nextjs-website/src/lib/cmsApi.ts [hunk 2]
```diff
@@ -103,6 +106,19 @@ export const getTutorialsProps = async (productSlug?: string) => {
   }
 };
 
+export const getTutorialListPagesProps = async () => {
+  const {
+    config: { FETCH_FROM_STRAPI: fetchFromStrapi },
+  } = buildEnv;
+
+  if (fetchFromStrapi) {
+    const strapiTutorialList = await fetchTutorialList(buildEnv);
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/cmsApi.ts:115]: Considering that `makeTutorialListPagesProps` inputs are lists of pages (`strapiTutorialListPages` and `staticTutorialListPages`) and not lists of tutorials, maybe you should rename this constant as follows:
```suggestion
    const strapiTutorialListPages = await fetchTutorialList(buildEnv);
```
and rename tutorialLists constant as well.

I would also consider renaming `fetchTutorialList` method in api.ts adopting the same logic.

File: apps/nextjs-website/src/lib/strapi/__tests__/tutorialList.test.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
+import * as E from 'fp-ts/lib/Either';
+import { guideImageJson } from './guides.test';
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/__tests__/tutorialList.test.ts:2]: We could create a file dedicated to test's features like images, icons, products, meta, etc.

File: apps/nextjs-website/src/lib/tutorialListPages.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      const product = mergeProductWithStaticContent(
+        attributes.product.data.attributes
+      );
+      const tutorials: readonly Tutorial[] = attributes.tutorials.data.map(
+        ({ attributes }) => ({
+          name: attributes.title,
+          path: `/${product.slug}/tutorials/${attributes.slug}`,
+          title: attributes.title,
+          publishedAt: attributes.publishedAt,
+          showInOverview: false,
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/tutorialListPages.ts:23]: I think we don't need this property anymore

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/lib/strapi/tutorialListCodec.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import * as t from 'io-ts/lib';
+import * as qs from 'qs';
+import { fetchFromStrapi } from './fetchFromStrapi';
+import { BaseProductCodec } from './codecs/ProductCodec';
+import { PaginationCodec } from './codecs/PaginationCodec';
+import { BaseTutorialCodec } from './tutorial';
+import { BannerLinkCodec } from './codecs/BannerLinkCodec';
+
+const StrapiTutorialListCodec = t.strict({
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/tutorialListCodec.ts:9]:
```suggestion
const StrapiTutorialListPageCodec = t.strict({
```

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/lib/tutorialListPages.ts (hunk 4 above)
Comment (MarcoPonchia) [reply apps/nextjs-website/src/lib/tutorialListPages.ts:23]: I think `showInOverview` could be removed once static content is no loger used but for now i think is necessary

**Review by marcobottaro — APPROVED** [review]

**Pull request merged**